### PR TITLE
ci/examples: Improve CI config

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/actions/verify-examples/setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+runs:
+  using: composite
+  steps:
+  - id: url
+    run: |
+        echo "base=https://storage.googleapis.com/envoy-${{ github.event_name == 'pull_request' && 'pr' || 'postsubmit' }}/${REF:0:7}/docker" \
+            >> "$GITHUB_OUTPUT"
+    env:
+      REF: ${{ inputs.ref || '5d58edacad73d5ac01edea989fd81c9d817987e8' }}
```
Comment (phlax) [inline .github/actions/verify-examples/setup/action.yml:10]: fix this

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

File: .github/actions/verify/examples/setup/action.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+runs:
+  using: composite
+  steps:
+  - id: url
+    run: |
+        echo "base=https://storage.googleapis.com/${BUCKET}/${REF:0:7}/docker" \
+            >> "$GITHUB_OUTPUT"
+    env:
+      BUCKET: envoy-${{ github.event_name == 'pull_request' && 'pr' || 'postsubmit' }}
+      REF: ${{ inputs.ref }}
```
Comment (phlax) [inline .github/actions/verify/examples/setup/action.yml:10]: i think this needs to be explicit

**Comment by phlax** [comment]
@wbpcode i spotted an issue with one of the template vars that should be fixed

also flagging that when this lands it will fail initially as the workflow that azp triggers will have moved

#28459 should fix that, the only way to avoid this would have been to first copy the wf, and then change, and then remove old etc

File: .github/workflows/envoy-publish.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          run_pre_with: '{"ref": "${{ inputs.ref }}"}'
+          env: |
+            export NO_BUILD_SETUP=1
+    uses: ./.github/workflows/_ci.yml
+    name: CI ${{ matrix.target }}
+    with:
+      target: ${{ matrix.target }}
+      rbe: ${{ matrix.rbe }}
+      managed: ${{ matrix.managed }}
+      cache_build_image: ${{ needs.env.outputs.build_image_ubuntu }}
```
Comment (phlax) [inline .github/workflows/envoy-publish.yml:70]: make configurable

**Comment by phlax** [comment]
reWIPping while i test/fix

**Comment by phlax** [comment]
should be fixed

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**